File: peek_plugin_diagram/_private/tuples/GridCacheIndexTuple.py
from datetime import datetime
from typing import Dict

from peek_plugin_diagram._private.PluginNames import diagramTuplePrefix
from vortex.Tuple import addTupleType, TupleField, Tuple


@addTupleType
class GridCacheIndexTuple(Tuple):
    __tupleType__ = diagramTuplePrefix + "GridCacheIndexTuple"

    # Improve performance of the JSON serialisation
    __rawJonableFields__ = ('data',)

    data: Dict[str, str] = TupleField()

File: peek_plugin_diagram/_private/both-app/tuples/GridCacheIndexTuple.ts
import { addTupleType, Tuple } from "@synerty/vortexjs";
import { diagramTuplePrefix } from "@peek/peek_plugin_diagram/_private";


// ----------------------------------------------------------------------------
/** Grid Cache Index
 *
 * The index is probably a terrible name.
 *
 * This tuple stores the updateDate for all grids cached in the database.
 *
 */
@addTupleType
export class GridCacheIndexTuple extends Tuple {
    public static readonly tupleName = diagramTuplePrefix + "GridCacheIndexTuple";

    // Improve performance of the JSON serialisation
    protected _rawJonableFields = ['data'];

    data: { [gridKey: string]: string } = {};

    constructor() {
        super(GridCacheIndexTuple.tupleName)
    }


}

